### PR TITLE
OBPIH-3355; OBPIH-3258: Add cancel adjustment functionality and refre…

### PR DESCRIPTION
**Pull request by DominikaHatala**
…sh adjustments tab when delete item

### CODE REVIEW

**Comment by DominikaHatala** [comment]
@awalkowiak @jmiranda This one requries database migration - new boolean field "isCanceled" was added to order adjustment.

File: grails-app/views/purchaseOrder/_showOrderItems.gsp [hunk 1]
```diff
@@ -394,6 +392,17 @@
           });
         }
 
+        function getTotalAdjustments() {
```
Comment (jmiranda) [inline grails-app/views/purchaseOrder/_showOrderItems.gsp:395]: This doesn't feel very DRY :) 

We could probably add a utility library for methods like this. I have the same or very similar code in a few places in _productDetails.gsp. Perhaps we could make one single function in a utils.js library that does the following

```
function refreshDataPoint(url, data, target, digits) { 
    $.ajax({
        url:url,
        data: data,
        success: function(data, textStatus){
              target.html(parseFloat(data).toFixed(digits));
        }
    });
}
```

And then on pages like this we'd just use the library method.
```
function getTotalAdjustments() { 
   var target = $("#totalAdjustments");
   var data = { orderId: $("#orderId").val() }
   var url = "${g.createLink( controller:'order', action:'getTotalAdjustments')}"
   refreshDataPoint(url, data, target, 2)
}
```

In this case, I understand this doesn't really save us all that much, but it'll be much easier to troubleshoot and extend if it's all in one place.

**Review by jmiranda — CHANGES_REQUESTED** [review]
LGTM. just take a look at the comment about refactoring.

File: grails-app/views/purchaseOrder/_showOrderItems.gsp (hunk 1 above)
Comment (DominikaHatala) [reply grails-app/views/purchaseOrder/_showOrderItems.gsp:395]: @jmiranda I think it's a good idea. I'll create a new ticket for that. Does that sounds good?

**Comment by DominikaHatala** [comment]
@jmiranda @awalkowiak  I have created new ticket (https://pihemr.atlassian.net/browse/OBPIH-3386) for adding utils library so I think we can merge this one.

**Review by awalkowiak — APPROVED** [review]

**Comment by awalkowiak** [comment]
LGTM I am merging it because I need the code from this pr

**Pull request merged**